packages/react/src/Button/ButtonBase.tsx

```diff
@@ -72,7 +72,8 @@ const ButtonBase = forwardRef(
         if (
           innerRef.current &&
           !(innerRef.current instanceof HTMLButtonElement) &&
-          !((innerRef.current as unknown) instanceof HTMLAnchorElement)
+          !((innerRef.current as unknown) instanceof HTMLAnchorElement) &&
+          !((innerRef.current as HTMLElement).tagName === 'SUMMARY')
```

Comment:
Removes console.warn for [components-details--default](https://primer.style/react/storybook/?path=/story/components-details--default)